### ISSUE TITLE
Parse throws DecodingError.missingKey for key "Message" 

### ISSUE THREAD

**Issue by tapwork** [contentful/contentful.swift#65]
We created some models in Contentful but the SDK always breaks / throws at

```
func parse(_ json: Any, _ keyPath: KeyPath) throws -> Any {
    var currentDict = json
    
    for (index, key) in keyPath.keys.enumerated() {
        guard let result = try NSDictionary.decode(currentDict)[key] else {
            let currentPath = keyPath.keys[0 ..< index]
            let metadata = DecodingError.Metadata(path: Array(currentPath), object: currentDict, rootObject: json)
            throw DecodingError.missingKey(key, metadata)
        }
        
        currentDict = result
    }
    
    return currentDict
}
```
 
`throw DecodingError.missingKey(key, metadata)` key: `message`

We are using 
```
Contentful (0.3.1):
    - Decodable (~> 0.5)
    - Interstellar (~> 2.0.0)
```

**Comment by loudmouth** [comment]
Hi @tapwork,

Can you please upgrade to the [latest version of contentful.swift (version 0.4.1)](https://github.com/contentful/contentful.swift/releases/tag/0.4.1) and see if you're still having the same issue?

Decodable is no longer a dependency of Contentful anymore. It has been replaced with [ObjectMapper](https://github.com/Hearst-DD/ObjectMapper)

**Comment by tapwork** [comment]
I even switched to the `ContentfulPersistence` and https://github.com/contentful/contentful.swift/releases/tag/0.4.1
But there I get lots of `MapError` (throw)

<img width="1036" alt="screen shot 2017-05-31 at 09 00 19" src="https://cloud.githubusercontent.com/assets/397342/26619709/eaf84d8e-45df-11e7-8fd7-610e4e393f1b.png">

My mapping looks as follows:
```
        synchronizer.mapAssets(to: Asset.self)
        synchronizer.map(contentTypeId: "teaser", to: FeedTeaser.self)
``` 

I am using all the properties for CoreData I have in the Contentful CMS.
But the mapping issues occur with the following the keys:
`locale`, `contentType.sys.id`, `createdAt`, `updatedAt`, `revision`, `message`

Those properties are not part of my model. They look more like the sys properties

The mapping for entities itself seems to work.

**Comment by loudmouth** [comment]
I was not aware you were using ContentfulPersistence as well. Make sure you have a property on your `FeedTeaser` that is exactly like the following:

```
@NSManaged var id: String?
```

If you still have trouble, please open a support ticket with [support.contentful.com](https://support.contentful.com). When doing so, please link this Github issue and mention that the iOS developer sent you to the support channel so that the ticket will be assigned to me by our support team.

Also, send along the following information:
- A copy of your `FeedTeaser` class.
- If you are using Cocoapods or Carthage for installation.
- The block of code where you call `sync()` on your `ContentfulSynchronizer` instance.

Thanks.

**Comment by tapwork** [comment]
Thanks for your fast answer. We just switched to `ContentfulPersistence`
I have the `id` key as well. But still...I will send a detailed description to support.
But nevertheless how do I get the `NSManagedObject` models after fetching entries?
```
self.client.fetchEntries(matching: ["content_type": contentType.rawValue]) { result in
            switch result {
            case .success(let arrayResponse):
                // how to get the NSManagedObject objects here from arrayResponse
            case .error(let error):
                completion(.failure(ContentfulResultError.failedResponse(error)))
            }
        }
```

**Comment by loudmouth** [comment]
Hi @tapwork [contentful-persistence.swift](https://github.com/contentful-persistence.swift) is only designed to specifically work with the [`/sync` endpoint](https://www.contentful.com/developers/docs/references/content-delivery-api/#/reference/synchronization) of the Content Delivery API. If you are using the regular `/entries` or `/assets` endpoints, you must map your entries to `NSManagedObject` instances on your own.


In the comment you left above `// how to get the NSManagedObject objects here from arrayResponse`, this is the point where you would perform your mapping and save to CoreData.

**Comment by loudmouth** [comment]
As an added note, if you are only using these endpoints, the persistence library will be quite useless for you, and you may want to consider pruning it from  your project.

**Comment by tapwork** [comment]
is this the proper way to fetch from /sync endpoint?
```
self.client.fetchEntries(["content_type": contentType.rawValue]).1.next {
            print($0)
        }
```
got this from https://www.contentful.com/developers/docs/ios/tutorials/using-delivery-api-with-swift/

**Comment by loudmouth** [comment]
The correct way to interact with the sync endpoint is indeed the later snippet you posted. It is not necessary to use the persistence library to do this however. You can see the reference documentation for the sync methods on the `Client` type of contentful.swift [here](https://contentful.github.io/contentful.swift/Classes/Client.html#/s:FC10Contentful6Client11initialSyncFT8matchingGVs10DictionarySSP__10completionFGO12Interstellar6ResultCS_9SyncSpace_T__GSqCSo18URLSessionDataTask_)—as a side note, the reference docs are fully compatible with Xcode's documentation system, so you should be able to navigate through the SDK in Xcode as well.

To properly use [contentful-persistence.swift](https://github.com/contentful/contentful-persistence.swift), you would do the following:

```
self.coreDataStore = CoreDataStore(context: self.managedObjectContext) // You're responsible for creating you managedObjectContext
let synchronizer = ContentfulSynchronizer(client: self.client, persistenceStore: self.coreDataStore)
synchronizer.mapAssets(to: Asset.self)
synchronizer.map(contentTypeId: "teaser", to: FeedTeaser.self)

// Do the sync operation with the API.
synchronizer.sync() { success in 
  // Now you can fetch from CoreData
  let teasers: [FeedTeaser] = self.coreDataStore.fetchAll(type: FeedTeaser.self, predicate: NSPredicate(value: true))
}


```

**Comment by loudmouth** [comment]
Fixed in #70
